tests/cypress/plugins/index.js

```diff
@@ -41,5 +41,11 @@ module.exports = (on, config) => {
         }
         return launchOptions;
     });
+
+    on('after:spec', (spec, results) => {
+        if (results.stats.failures === 0 && results.video) {
```

Comment:
Documentation recommends to check `results` itself: https://docs.cypress.io/api/plugins/after-spec-api

```suggestion
        if (results?.stats.failures === 0 && results?.video) {
```